### PR TITLE
Allow to run with restricted pod security standards

### PR DESCRIPTION
**Pull request by ramondeklein**
It's possible to [enforce pod security standards](https://kubernetes.io/docs/concepts/security/pod-security-standards) by either [adding labels](https://kubernetes.io/docs/tasks/configure-pod-container/enforce-standards-namespace-labels/) or by [configuring the built-in admission controller](https://kubernetes.io/docs/tasks/configure-pod-container/enforce-standards-admission-controller/). There are three modes available: [**privileged**](https://kubernetes.io/docs/concepts/security/pod-security-standards/#privileged) (default), [**baseline**](https://kubernetes.io/docs/concepts/security/pod-security-standards/#baseline) and [**restricted**](https://kubernetes.io/docs/concepts/security/pod-security-standards/#restricted). It looks like both the operator and tenants work in _restricted_ mode, but it requires that all `containerSecurityContext` values need the following settings:
```yaml
allowPrivilegeEscalation: false
capabilities:
  drop:
    - ALL
seccompProfile:
  type: RuntimeDefault
```
This setting can be found in the following places:
- [`operator.containerSecurityContext`](https://github.com/minio/operator/blob/aa1337cfc75a55b0c86ddeaaaf6814dbaf66e86b/helm/operator/values.yaml#L94) (Operator helm chart)
- [`console.containerSecurityContext`](https://github.com/minio/operator/blob/aa1337cfc75a55b0c86ddeaaaf6814dbaf66e86b/helm/operator/values.yaml#L279) (Operator helm chart)
- [`tenant.pools[*].containerSecurityContext`](https://github.com/minio/operator/blob/aa1337cfc75a55b0c86ddeaaaf6814dbaf66e86b/helm/tenant/values.yaml#L202) (Tenant helm chart)

This PR adds these values to the Helm charts `values.yaml` files to ensure that the deployment/stateful-set is also allowed when pod security is set to `restricted`.

The console is also updated to ensure that it also deploys `tenant` resources with these values set. All CI/CD tests are updated so they run with the `operator` namespace in `restricted` mode.

### CODE REVIEW

File: helm/operator/values.yaml [hunk 1]
```diff
@@ -279,6 +285,12 @@ console:
   containerSecurityContext:
     runAsUser: 1000
     runAsNonRoot: true
+    allowPrivilegeEscalation: false
```
Comment (ramondeklein) [inline helm/operator/values.yaml:288]: All other container security contexts also use `runAsGroup: 1000`, but the console doesn't. Should we add that too?

**Review by pjuarezd — APPROVED** [review]

File: helm/operator/values.yaml (hunk 1 above)
Comment (pjuarezd) [reply helm/operator/values.yaml:288]: I think that would be good, yes please @ramondeklein

Comment (ramondeklein) [reply helm/operator/values.yaml:290]: Fixed with 1611fa2735aa6ebdd50fce3f9a75e30ffde5db98.

**Review by pjuarezd — APPROVED** [review]

**Review by cniackz — DISMISSED** [review]
I have tested the changes, and I can deploy tenants and place objects with these values. This modification appears promising to me, especially when the `pod-security.kubernetes.io/enforce=restricted` label is utilized in the namespace.

**Comment by cniackz** [comment]
If the decommissioning test fails, I believe we can disregard this failure as it is not relevant. However, the improvements needed for the test itself should be addressed in a new PR.

**Review by pjuarezd — APPROVED** [review]

**Review by cniackz — DISMISSED** [review]

**Comment by cniackz** [comment]
@ramondeklein approved but let's resolve conflicts

**Comment by cniackz** [comment]
Great, currently reviewing and testing!

File: resources/base/namespace.yaml [hunk 2]
```diff
@@ -2,3 +2,10 @@ apiVersion: v1
 kind: Namespace
 metadata:
   name: minio-operator
+  labels:
+    pod-security.kubernetes.io/enforce: restricted
+    pod-security.kubernetes.io/enforce-version: latest
+    pod-security.kubernetes.io/audit: restricted
+    pod-security.kubernetes.io/audit-version: latest
+    pod-security.kubernetes.io/warn: restricted
+    pod-security.kubernetes.io/warn-version: latest
```
Comment (cniackz) [inline resources/base/namespace.yaml:11]: Could you please add a newline at the end of the file?

```suggestion
    pod-security.kubernetes.io/warn-version: latest

```

**Review by cniackz — APPROVED** [review]
All good, just a minor change requested. And thank you for adding the test in the Helm chart.

**Review by pjuarezd — APPROVED** [review]

**Review by cniackz — APPROVED** [review]

**Pull request merged**